engine/src/main/java/org/terasology/rendering/dag/stateChanges/SetCullFaceMode.java

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright 2016 MovingBlocks
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.terasology.rendering.dag.stateChanges;
+
+import org.lwjgl.opengl.GL11;
+import static org.lwjgl.opengl.GL11.GL_BACK;
+import static org.lwjgl.opengl.GL11.GL_FRONT;
+import static org.lwjgl.opengl.GL11.GL_FRONT_AND_BACK;
+import org.terasology.rendering.dag.RenderPipelineTask;
+import org.terasology.rendering.dag.StateChange;
+import org.terasology.rendering.dag.tasks.SetCullFaceModeTask;
+
+/**
+ * TODO: Add javadocs
+ */
+public final class SetCullFaceMode implements StateChange {
+
+    private static SetCullFaceMode defaultInstance = new SetCullFaceMode(GL_BACK); // also specified in OpenGL documentation
+    private SetCullFaceModeTask task;
+    private int mode;
+
+    public SetCullFaceMode(int mode) {
+        this.mode = mode;
+        validate();
+    }
+
+    private void validate() {
```

Comment:
Nah. The intent of the code it's clear: constructor input validation. Because it's so simple it can stay in the constructor without a method wrapping it. I have not read those to books, in my library I have Code Complete for those kind of issues. My understanding of it is that we need to strike a balance: too large and complex methods are hard to read, but segmenting the code into very small methods is just as bad. So, I'd still remove validate() as this is a very trivial validation.